src/rapid_pro_to_engagement_db/rapid_pro_to_engagement_db.py

```diff
@@ -174,15 +180,34 @@ def sync_rapid_pro_to_engagement_db(rapid_pro, engagement_db, uuid_table, rapid_
     # (If the cache or a contacts file for this workspace don't exist, `contacts` will be `None` for now)
     contacts = _get_contacts_from_cache(cache)
 
-    dataset_to_sync_stats = dict()  # of '{flow_name}.{flow_result_field}' -> RapidProToEngagementDBSyncStats
-    for flow_config in rapid_pro_config.flow_result_configurations:
-        sync_stats = RapidProToEngagementDBSyncStats()
+    # Check the configs are the same before proceeding with cached data
+    cached_flow_result_configs = _get_flow_result_configs_from_cache(cache)
+    if cached_flow_result_configs is None and cache is not None and not dry_run:
+        cache.set_flow_result_configs(rapid_pro_config.flow_result_configurations)
+    else:
+        cached_flow_result_configs = [d.to_dict() for d in cached_flow_result_configs]
+        updated_flow_result_configs = [config for config in rapid_pro_config.flow_result_configurations \
+                                        if config.to_dict() not in cached_flow_result_configs]            
+        if len(updated_flow_result_configs) > 0:
```

Comment:
This covers the case I was worried about, which is that if you added a new row to the flow configs it wouldn't be obvious from the structure of the flow configurations that a cache reset was needed. Can you add a todo for the case it doesn't yet cover, which is if you remove a flow configuration, re-run, then add the same configuration back again, which might happen if you were switching between branches.